Modules/arraymodule.c

```diff
@@ -332,34 +332,57 @@ II_getitem(arrayobject *ap, Py_ssize_t i)
 }
 
 static int
+get_int_unless_float(PyObject **vv)
```

Comment:
I would just return `PyObject *` (`NULL` if error). This simplifies the code and perhaps makes easier changing it for using `__index__`.